### ISSUE TITLE
Editing custom units is impossible because the combo box is cleared when typing

### ISSUE THREAD

**Issue by rovo89** [grafana/grafana#82896]
### What happened?

It's nice that we can use custom units, but there's an annoying bug in the options control that makes it impossible to edit the details after entering them. As soon as I start typing anything into the "Unit" combo box, it clears the previous content, so I have to write the whole string from scratch.

### What did you expect to happen?

I would like to fine-tune my custom unit (e.g. adjusting the date format), so instead of clearing/overwriting everything, it should just insert the newly typed characters like in a normal textbox.

A good middle ground would be selecting the whole string when the combo box comes into focus. If I then just start typing, I expect the previous content to be overwritten. But if I place the cursor at the end, it should just append the new text.

### Did this work before?

Not that I'm aware of.

### How do we reproduce it?

1. Open edit mode on any panel, e.g. https://play.grafana.org/d/000000016/1-time-series-graphs?orgId=1&editPanel=1
2. Under "Unit", type something like "suffix:ab" and hit Enter. Observe that the values are shown as "20 ab" etc.
3. Place the cursor in the "Unit" combo box again by clicking behind the existing string (so it will be like "suffix:ab|").
4. Press "c" on the keyboard.

Expected: The content changes to "suffix:abc".
Actual: The content changes to "c".

Same wrong behavior if the cursor is places in the beginning or middle of the string in step 3, or if any text has been selected.

### Environment (with versions)?

Grafana: v10.4.0-65039 (da894994d4) --> Playground
OS: Windows 11
Browser: Chrome 121.0.6167.161

### Grafana platform?

None

**Comment by JoaoSilvaGrafana** [comment]
Hey! 👋  Thanks for raising this! I was able to reproduce it and figured out the issue is on the `Cascader` component. What happens is that when you open the `Cascader`, it renders the `rc-cascader` component, but as you start typing something it switches it to a regular `Select` and just loses what was on the input.

I've tried to fix this quickly by having the Select input being controlled, but that makes some parts of this interaction not ideal, for instance selecting the whole input and then typing doesn't replace the whole content, just adds what you typed to the end, so I'd like to get some input on if this is a good solution or not before moving ahead with the fix

**Comment by rovo89** [comment]
@JoaoSilvaGrafana Thanks for the fix! I noticed a bit late that the official v11.0.0 (277ef258d4) is out. Just tested it there and it partially fixed the issue.

https://github.com/grafana/grafana/assets/1573299/c97cde3a-fa52-4de4-a031-9b9cc082461b

In the video you see:
- Coming from unfocused state, placing the cursor via mouse in the middle of the text and then typing "x" places the new text at the end, rather than where the cursor was.
- While editing, I can't click elsewhere in the text to place the cursor. Instead, it leaves edit mode.
- Moving the cursor with the keyboard does work, so with that workaround I can edit the unit without completely retyping it.

---

However, I also checked in the playground, which is Grafana v11.1.0-70958 (8c2a382788) right now, and there it seems that the fix is completely missing (regression):

https://github.com/grafana/grafana/assets/1573299/7888264c-1449-4143-9ca3-5474c7bed957
